dataquality/loggers/model_logger/image_classification.py

```diff
@@ -33,3 +35,45 @@ def __init__(
             epoch=epoch,
             inference_name=inference_name,
         )
+
+    def _filter_duplicate_ids(self) -> None:
+        """To avoid duplicate ids, when augmentation is used.
+        Filter out duplicate ids in the batch. This is done by keeping track of
+        the ids that have been observed in the current epoch in the config"""
+        observed_ids = self.logger_config.observed_ids[str(self.epoch)]
+        unique_ids = set(self.ids).difference(observed_ids)
+        observed_ids.update(unique_ids)
+        # If there are duplicate ids, filter out the duplicates
+        if len(self.ids) > len(unique_ids):
+            unique_indices = [list(self.ids).index(id) for id in unique_ids]
+            if len(self.embs) > 0:
+                self.embs = np.array(self.embs)[unique_indices]
+            if len(self.probs) > 0:
+                self.probs = np.array(self.probs)[unique_indices]
+            if len(self.ids) > 0:
+                self.ids = np.array(self.ids)[unique_indices]
+
+    def _get_data_dict(self) -> Dict[str, Any]:
```

Comment:
you can refactor this a bit, to make it utilize the parent function a bit more
```suggestion
    def _get_data_dict(self) -> Dict[str, Any]:
        self._filter_duplicate_ids()
        if len(self.ids) != len(set(self.ids)):
            raise ValueError(
                f"Duplicate ids found in batch: {self.ids}. "
                "Please make sure that each sample has a unique id."
            )
        return super()._get_data_dict()
```